lib/backend/pgbk/put_batch.go

```diff
@@ -0,0 +1,81 @@
+/*
+ * Teleport
+ * Copyright (C) 2025  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package pgbk
+
+import (
+	"context"
+	"slices"
+
+	"github.com/gravitational/trace"
+	"github.com/jackc/pgx/v5/pgtype/zeronull"
+
+	"github.com/gravitational/teleport/lib/backend"
+	pgcommon "github.com/gravitational/teleport/lib/backend/pgbk/common"
+)
+
+const (
+	defaultUpsertBatchChunk = 300
+	putBatchStmt            = `
+INSERT INTO kv (key, value, expires, revision)
+SELECT k, v, e, r FROM UNNEST(
+	$1::bytea[],
+	$2::bytea[],
+	$3::timestamptz[],
+	$4::uuid[]
+) AS t(k, v, e, r)
+ON CONFLICT (key) DO UPDATE
+SET
+	value    = EXCLUDED.value,
+	expires  = EXCLUDED.expires,
+	revision = EXCLUDED.revision;
+`
+)
+
+// PutBatch puts multiple items into the backend in a single transaction.
+func (b *Backend) PutBatch(ctx context.Context, items []backend.Item) ([]string, error) {
+	if len(items) == 0 {
+		return nil, trace.BadParameter("at least one item must be provided")
+	}
```

Comment:
I don't see why we have to force the caller to do the check.

```suggestion
	if len(items) == 0 {
		return nil, nil
	}
```